big_tests/tests/mod_http_notification_SUITE.erl

```diff
@@ -151,44 +125,94 @@ simple_message_no_listener(Config) ->
 simple_message_failing_listener(Config) ->
     do_simple_message(Config, <<"Hi, Failing!">>).
 
-do_simple_message(Config, Msg) ->
-    BobJid = escalus_users:get_jid(Config, bob),
-
+do_simple_message(Config0, Msg) ->
+    Config = escalus_fresh:create_users(Config0, [{alice, 1}, {bob, 1}]),
     %% Alice sends a message to Bob, who is offline
-    escalus:story(Config, [{alice, 1}],
-        fun(Alice) -> escalus:send(Alice, escalus_stanza:chat_to(BobJid, Msg)) end),
-
+    {ok, Alice} = escalus_client:start(Config, alice, <<"res1">>),
+    escalus:send(Alice, escalus_stanza:presence(<<"available">>)),
+    BobJid = escalus_users:get_jid(Config, bob),
+    Stanza = escalus_stanza:chat_to(BobJid, Msg),
+    escalus:send(Alice, Stanza),
+    escalus_client:stop(Config, Alice),
     %% Bob logs in
-    Bob = login_send_presence(Config, bob),
-
+    {ok, Bob} = escalus_client:start(Config, bob, <<"res1">>),
+    escalus:send(Bob, escalus_stanza:presence(<<"available">>)),
     %% He receives his initial presence and the message
     Stanzas = escalus:wait_for_stanzas(Bob, 2),
-%%  ct:pal("Stanzas:~p", [Stanzas]),
     escalus_new_assert:mix_match([is_presence, is_chat(Msg)], Stanzas),
     escalus_client:stop(Config, Bob).
 
-
-%%%===================================================================
-%%% Custom predicates
-%%%===================================================================
-
-is_chat(Content) ->
-    fun(Stanza) -> escalus_pred:is_chat_message(Content, Stanza) end.
-
 %%%===================================================================
 %%% Helpers
 %%%===================================================================
 
+get_http_request() ->
+    Key = got_http_request,
+    mongoose_helper:wait_until(
+      fun() -> 1 =:= length(ets:lookup(?ETS_TABLE, Key)) end,
+      true, #{name => missing_request}),
+    [Bins] = lists:map(fun({_, El}) -> El end, ets:lookup(?ETS_TABLE, Key)),
+    ets:delete(?ETS_TABLE, Key),
+    Bins.
+
 login_send_presence(Config, User) ->
     Spec = escalus_users:get_userspec(Config, User),
     {ok, Client} = escalus_client:start(Config, Spec, <<"dummy">>),
     escalus:send(Client, escalus_stanza:presence(<<"available">>)),
     Client.
 
-get_prefix(mod_http_notification_tests) ->
+is_chat(Content) ->
+    fun(Stanza) -> escalus_pred:is_chat_message(Content, Stanza) end.
+
+get_prefix(no_prefix) ->
     "/";
-get_prefix(mod_http_notification_tests_with_prefix) ->
+get_prefix(with_prefix) ->
     "/prefix";
 get_prefix(Config) ->
     GroupName = proplists:get_value(name, proplists:get_value(tc_group_properties, Config)),
     get_prefix(GroupName).
+
+start_pool() ->
+    HTTPOpts = [{server, http_notifications_host()}],
+    PoolOpts = [{strategy, available_worker}, {workers, 1}],
```

Comment:
I thought so because tests are run in sequence, so there's no need for so many listeners. But perhaps it was mostly because log messages from gun crashing were multiplied by ten 😅  Will fix anyway 🙂

Reply:
I would go for more because it may show some problems, like "Why are there so many crash logs?! We should clean it up!" and because it is probably the way the pool will be ran, even if the actual number of requests will be really low.